Changed treatment of gravity in poromechanics kernels

francoishamon opened the pull request:
This PR changes the formulation of the body force and pressure terms in the poromechanics kernels (single-phase and multiphase) to move away from the delta formulation (delta pressure, delta body force, stress interpreted as a delta from an initial stress). I am going to come up with a simple validation test that can also be added to the integrated tests, and then ask for a review.

The PR also removes some constitutive fields that are no longer needed after the change: 
- `initialDensity` in `SingleFluidBase`
- `initialTotalMassDensity` in `MultiFluidBase`

Resolves #2041 

Requires rebaseline

CusiniM commented:
I would use the case created by @cssherman in https://github.com/GEOSX/GEOSX/issues/2041 as an integratedTest.

francoishamon commented:
> I would use the case created by @cssherman in #2041 as an integratedTest.

Sorry for the delay, I just added the test created by Chris to this PR.

The change proposed in this PR will also affect no-gravity poromechanics simulations because it changes the formulation of the total stress in the code. Specifically, the pressure term is no longer a pressure increment (but the actual pressure value) and the body force term is no longer an increment either. 

So, in the poromechanics simulations with this PR, the `stress` variable of the xml file is an effective stress that does not include the pressure term, and we have to initialize it properly when the simulation has either gravity and/or a non-zero initial pressure to start from an equilibrium. In our integrated tests and validation tests, I added this effective stress initialization where it was missing to make sure that we don't have numerical diffs. The only thing that I was not able to do is [changing this script](https://geosx-geosx.readthedocs-hosted.com/en/latest/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/deviatedPoroElasticWellbore/deviatedPoroElasticWellboreExample2Figure.py) and its input data because it was generated outside GEOSX. 

The main advantage is that the stress computed by GEOSX is not anymore a delta with respect an equilibrium configuration, but now has a meaningful value that users can use in their analyses. But I may be wrong. Is everyone on board with the change?

francoishamon commented:
If everyone is on board with the changes, this PR is ready to be merged.

jhuang2601 commented on src/docs/sphinx/advancedExamples/validationStudies/faultMechanics/faultVerification/Example.rst:
```diff
@@ -2,7 +2,7 @@
 
 ###########################################################
-Verification of Induced Stresses Along a Fault
+Verification of Induced Sresses Along a Fault
```
```suggestion
Verification of Induced Stresses Along a Fault
```

francoishamon replied:
My bad, this change was not intentional. It is fixed now.

jhuang2601 commented on src/coreComponents/constitutive/solid/PorousSolid.hpp:
```diff
@@ -378,61 +368,52 @@ class PorousSolidUpdates : public CoupledSolidUpdates< SOLID_TYPE, BiotPorosity,
                          real64 ( & dBodyForce_dPressure )[3],
                          real64 ( & dBodyForce_dComponents )[3][NUM_MAX_COMPONENTS] ) const
   {
     computeBodyForce( solidDensity,
```
`bodyForce` is induced by gravity only. How about tectonic stress/strain? Any plan for these?

francoishamon replied:
For stress/strain, the formulation that we have in place works well and does not require any change. For field-scale models, we still have to test the poromechanical initialization from a given regional stress, but my understanding is that we have everything we need now (I may be wrong :) ).

CusiniM reviewed (approved):
I personally prefer it this way. Isn't it also more consistent with what we do for the `stress` in general?

castelletto1 reviewed (approved):
Looks good to me. Thanks!

francoishamon commented:
> I personally prefer it this way. Isn't it also more consistent with what we do for the `stress` in general?

Yes, with the previous formulation, stress had to be interpreted as a delta stress from an initial equilibrium. Now, the initial stress has to be provided to GEOSX (we have used PR #1927 to compute the initial stress in the field-scale simulations) and the stress computed by GEOSX is no longer a delta wrt to the initial state. This is easier when users want to visualize / analyze / post-process the stress variable.
